utils/config.mli

```diff
@@ -153,13 +153,16 @@ val flambda : bool
 
 val spacetime : bool
         (* Whether the compiler was configured for Spacetime profiling *)
-val spacetime_call_counts : bool
+val disable_call_counts : bool
```

Comment:
You're changing the interface of `config.ml` (and also of `configure`) for no good reason that I can see. Why not just do what the title of the PR says, and change the default value of the existing variable? (and of course add a `-no-spacetime-call-counts` option to `configure`)

Reply:
Currently `with_spacetime_call_counts` means `with_spacetime && not disable_call_counts`. This isn't really consistent with other configure options like `disable_libunwind`, and it means that just changing the default of `with_spacetime_call_counts` to `true` would enable spacetime by default.

Reply:
In other words, I'm changing the name of this value in `config.ml` because I'm changing it's meaning, and I'm changing it's meaning to be consistent with both the other similar values in `config.ml` and with how it will appear in the arguments to `./configure.sh`.

Reply:
In `configure` yes, you're changing the meaning of the command-line option. In `config.ml` and in the C code, you have just added and removed negations, which (in my eye) is noise.

The variable itself never had the same meaning as the configure option, as can plainly be seen in `configure` itself.

Reply:
I'm also not a big fan of negative configuration variables, as these often lead to double-negated `#ifdef`s that make the code harder to read.

Reply:
> The variable itself never had the same meaning as the configure option, as can plainly be seen in configure itself.

I'm not sure I'm following this. The variable would previously bet set precisely when `spacetime-call-counts` was passed to configure, which would also always imply `with_spacetime`. This seems to have the same meaning between variable and configure option. Or do you mean that the implication that `with_spacetime` was true was never used -- as evidenced by my only needing to negate various tests in this patch.

I'm happy to change make the option positive -- but I would still rather rename it to something like `Config.enable_call_counts` in ml and `ENABLE_CALL_COUNTS` in c. In particular, I think that the `WITH_SPACETIME_CALL_COUNTS` is a bad name for the variable as it heavily implies that `WITH_SPACETIME` is true if it is true.

Reply:
What I mean is that the `configure` code says that `-spacetime-call-counts` means (with-spacetime AND with-spacetime-call-counts) which is obviously not the same as just with-spacetime-call-counts.

But anyway I'm OK with `enable_call_counts`. I don't much like `with_spacetime_call_counts` either.